### PR TITLE
Adding api

### PR DESCRIPTION
**Pull request by mzlatic99**
Pozdrav @kdelivuk @lukacevica @PetricMatej , molim za feedback :)

### CODE REVIEW

File: lib/screens/home-screen/widgets/homes_guests_love.dart [hunk 1]
```diff
@@ -1,68 +1,76 @@
...
-                fit: BoxFit.cover,
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Padding(
+          padding: const EdgeInsets.fromLTRB(20, 40, 20, 0),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Text(
```
Comment (lukacevica) [inline lib/screens/home-screen/widgets/homes_guests_love.dart]: Please separate list title + view more button to new widget.

File: lib/screens/home-screen/widgets/homes_guests_love.dart [hunk 2]
```diff
@@ -1,68 +1,76 @@
...
+              if (snapshot.connectionState == ConnectionState.waiting) {
+                return Center(
+                    child: CircularProgressIndicator(
+                  color: ThemeColors.mint500,
+                ));
+              }
+              List<Accommodation> accommodations = snapshot.data;
+              for (var accommodation in accommodations) {
+                _homesGuestsLoveList.add(HomeGuestsLoveCard(
+                    isHorizontalList: true,
```
Comment (lukacevica) [inline lib/screens/home-screen/widgets/homes_guests_love.dart:44]: You can pass the whole accommodation object here.

File: lib/screens/home-screen/widgets/popular_locations.dart [hunk 3]
```diff
@@ -1,59 +1,75 @@
...
-        ),
         Padding(
-          padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 10),
-          child: Column(
-            crossAxisAlignment: CrossAxisAlignment.start,
+          padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 20),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
             children: [
               Text(
```
Comment (lukacevica) [inline lib/screens/home-screen/widgets/popular_locations.dart]: Here you can use newly created widget for list title and view more button.

File: lib/screens/home-screen/widgets/popular_locations.dart [hunk 4]
```diff
@@ -1,59 +1,75 @@
...
+            if (snapshot.connectionState == ConnectionState.waiting) {
+              return Center(
+                  child: CircularProgressIndicator(
+                color: ThemeColors.mint500,
+              ));
+            }
+            List<Location> locations = snapshot.data;
+            for (var location in locations) {
+              _locationCardList.add(LocationCard(
+                  id: location.id,
```
Comment (lukacevica) [inline lib/screens/home-screen/widgets/popular_locations.dart]: You can pass the whole location object here.

File: lib/screens/homes_guests_love-screen/homes_guests_love_screen.dart [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                    if (snapshot.connectionState == ConnectionState.waiting) {
+                      return Center(
+                          child: CircularProgressIndicator(
+                        color: ThemeColors.mint500,
+                      ));
+                    }
+                    List<Accommodation> accommodations = snapshot.data;
+                    for (var accommodation in accommodations) {
+                      _homesGuestsLoveList.add(HomeGuestsLoveCard(
+                          isHorizontalList: false,
```
Comment (lukacevica) [inline lib/screens/homes_guests_love-screen/homes_guests_love_screen.dart:54]: You can pass the whole accommodation object here.

File: lib/screens/popular_locations_list-screen/popular_locations_list_screen.dart [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+              if (snapshot.connectionState == ConnectionState.waiting) {
+                return Center(
+                    child: CircularProgressIndicator(
+                  color: ThemeColors.mint500,
+                ));
+              }
+              List<Location> locations = snapshot.data;
+              for (var location in locations) {
+                _locationCardList.add(LocationCard(
+                    id: location.id,
```
Comment (lukacevica) [inline lib/screens/popular_locations_list-screen/popular_locations_list_screen.dart]: You can pass the whole location object here.

File: lib/shared/homes_guests_love_card.dart [hunk 7]
```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/material.dart';
+import '../../../theme.dart';
+
+class HomeGuestsLoveCard extends StatelessWidget {
+  final bool isHorizontalList;
+  final String id;
+  final String imageUrl;
+  final String title;
+  // final String shortDescription;
```
Comment (lukacevica) [inline lib/shared/homes_guests_love_card.dart]: Please delete unnecessary comments.

File: lib/shared/homes_guests_love_card.dart [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    // required this.capacity,
+    // required this.accommodationType,
+    // required this.freeCancelation,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> starsRating = [];
+    for (int i = 0; i < categorization; i++) {
+      starsRating.add(Icon(
```
Comment (lukacevica) [inline lib/shared/homes_guests_love_card.dart]: You can separate this to new widget. Widget that will return list of stars depending on categorization.

File: lib/screens/accommodation_details-screen/accommodation_details_screen.dart [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+              }
+              if (snapshot.connectionState == ConnectionState.waiting) {
+                return Center(
+                    child: CircularProgressIndicator(
+                  color: ThemeColors.mint500,
+                ));
+              }
+              List<Accommodation> accommodations = snapshot.data;
+              final List<Widget> starsRating = [];
+              for (int i = 0; i < 4; i++) {
```
Comment (lukacevica) [inline lib/screens/accommodation_details-screen/accommodation_details_screen.dart]: You can separate this to new widget. Widget that will return list of stars depending on categorization.

**Review by lukacevica — CHANGES_REQUESTED** [review]

**Comment by mzlatic99** [comment]
Hi @lukacevica, I'm asking for feedback on new changes :)

**Comment by lukacevica** [comment]
@mzlatic99  Nicely done! Your'e good to go and merge 😁

**Review by lukacevica — APPROVED** [review]

**Comment by mzlatic99** [comment]
Thank you😊

**Pull request merged**